assets/src/scripts/components/hydrograph/graph-brush.js

```diff
@@ -24,7 +25,14 @@ export const drawGraphBrush = function(container, store) {
 
         // Only about the main hydrograph when user is done adjusting the time range.
         if (event.sourceEvent.type === 'mouseup' || event.sourceEvent.type === 'touchend') {
-            store.dispatch(Actions.setHydrographXRange(brushRange.map(xScale.invert, xScale)));
+
+            const ts = currentVariablePointsByTsIdSelector('current')(store.getState());
```

Comment:
I think that this can be simplified considerably. You can get the domain of the brushRange (as was done before to set the hydrograph xrange. You can get the leftmost value of the range by using xScale.range()[0] and then use this to determine the offset. I think the method below will cause issues when the time series has masked data or missing value.

Reply:
I went ahead and used x.domain for the same purpose and everything seems to be working.